src/FSharp.Core/Linq.fs

```diff
@@ -740,7 +740,7 @@ module LeafExpressionConverter =
              let tupTy = 
                 let argTypes = args |> List.map (fun arg -> arg.Type) |> Array.ofList
                 if inp.Type.IsValueType then 
-                    Reflection.FSharpType.MakeStructTupleType(inp.Type.Assembly, argTypes)
+                    Reflection.FSharpType.MakeStructTupleType(argTypes)
```

Comment:
Confirmation based on the following:

The "else" branch here, refTuple generation, does not and did not accept "asm" attribute either.
In LINQ queries, ruf tuples are a lot more used and tested than struct tuples, and I did not find a filed issue about ref tuples in LINQ.

=> Changing struct tuples to behave the same like ref tuples, since they are part of runtime.

Reply:
Reverting back the usage from linq to use the old functionality.
There could be codebases using net46* or older, not having System.ValueType from the runtime and using it in a linq query.